lib/api_watch_diff.ml

```diff
@@ -15,6 +15,38 @@ let env_setup ~ref_sig ~curr_sig =
   let env = Env.add_signature ref_sig env in
   Env.add_signature curr_sig env
 
+let add_manifest_to_abstract_types ~ref_sig ~curr_sig =
+  let ref_types =
+    List.fold_left
+      (fun acc item ->
+        match item with
+        | Sig_type (id, decl, _, _)
+          when decl.type_kind = Type_abstract && decl.type_manifest = None ->
+            (Ident.name id, id) :: acc
+        | _ -> acc)
+      [] ref_sig
+  in
+  List.map
+    (fun item ->
+      match item with
+      | Sig_type (id, decl, rec_st, visibility)
+        when decl.type_kind = Type_abstract && decl.type_manifest = None -> (
+          let name = Ident.name id in
+          match List.find_opt (fun (n, _) -> n = name) ref_types with
+          | Some (_, ref_id) ->
+              let ref_path = Path.Pident ref_id in
+              let new_decl =
+                {
+                  decl with
+                  type_manifest =
+                    Some (Btype.newgenty (Tconstr (ref_path, [], ref Mnil)));
+                }
+              in
+              Sig_type (id, new_decl, rec_st, visibility)
```

Comment:
I think we can extract a function here, something like:
```ocaml
val set_type_equality : type_id:Ident.t -> type_declaration -> type_declaration
```

It's doing something non trivial and wrapping this into a named function would help understand what's going on!